lib/tokenable.rb

```diff
@@ -0,0 +1,11 @@
+module Tokenable
+  extend ActiveSupport::Concern
+
+  included do
+    before_create :generate_token
```

Comment:
Hehe this is what I did before changing my code to suit the Tokenable Concern (generating it on request).